components/ui/Button.tsx

```diff
@@ -1,76 +1,185 @@
-import { StyleSheet, TextStyle, TouchableOpacity, TouchableOpacityProps, ViewStyle } from "react-native";
+import {
+  StyleSheet,
+  TextStyle,
+  TouchableOpacity,
+  TouchableOpacityProps,
+  ViewStyle,
+} from "react-native";
 import CustomText from "@/components/ui/CustomText";
 import { ButtonVariant, SizeVariants } from "@/types";
 import { theme } from "@/theme";
 interface ButtonProps extends TouchableOpacityProps {
-    variant?: ButtonVariant;
-    textSize: SizeVariants;
-    title: string;
-    active?: boolean,
-    containerStyles?: ViewStyle,
-    textStyles?: TextStyle
+  variant?: ButtonVariant;
+  textSize?: SizeVariants;
+  title?: string;
+  active?: boolean;
+  containerStyles?: ViewStyle;
+  textStyles?: TextStyle;
+  disabled?: boolean;
+  fullWidth?: boolean;
+  children?: React.ReactNode;
 }
-export function CustomButton(
-    {
-        variant = 'primary',
-        onPress,
-        textSize,
-        textStyles = {},
-        containerStyles = {},
-        active,
-        title,
-        ...rest
-    }: ButtonProps): JSX.Element {
-
-    const variantStyles = {
-        primary: {
-            container: styles.primary,
-            text: styles.primaryVariantText,
+export function CustomButton({
+  variant = "default",
+  onPress,
+  textSize,
+  textStyles = {},
+  containerStyles = {},
+  active,
+  disabled,
+  children,
+  fullWidth,
+  title,
+  ...rest
+}: ButtonProps): JSX.Element {
+  const getVariantStyles = (
+    variant: string
+  ): { container: ViewStyle; text?: TextStyle } => {
+    if (variant === "primary") {
+      return {
+        container: styles.primary,
+        text: styles.primaryText,
+      };
+    }
+    if (variant === "secondary") {
+      return {
+        container: {
+          ...(fullWidth && styles.secondaryFullWidth),
+          ...styles.secondary,
         },
-        secondary: {
-            container: [styles.secondary, active && styles.secondaryActive],
-            text: [styles.secondaryVariantText, active && styles.secondaryActiveVariantText],
-        }
+        text: styles.secondaryText,
+      };
     }
+    if (variant === "outline") {
+      return {
+        container: styles.outline,
+        text: styles.outlineText,
+      };
+    }
+    if (variant === "destructive") {
+      return {
+        container: styles.destructive,
+        text: styles.descructiveText,
+      };
+    }
+    if (variant === "ghost") {
+      return {
+        container: styles.ghost,
+      };
+    }
+    if (variant === "default") {
```

Comment:
do we need the default variant? I think that whatever variant we consider to be the default should just be set as the default value of the variant prop and used as the fallback. For example, if the primary variant is meant to be the default, let's make it the default variant value, and so on.